pfl/internal/ops/pytorch_ops.py

```diff
@@ -296,12 +301,42 @@ def __init__(self, seed=None):
 
     def __enter__(self):
         if self._seed is not None:
-            self._saved_state = torch.random.get_rng_state()
+            self._saved_state = {"cpu": torch.random.get_rng_state()}
+            try:
```

Comment:
Should we not check if cuda or mps are available rather than having two try-except blocks, with at least one of these blocks generating a "Failed to get ..." message?

E.g. using `if torch.cuda.is_available():` and `if torch.backends.mps.is_available():`

Reply:
We don't want to print an info log with "Failed ..." for something expected. what about same control flow as in https://github.com/apple/pfl-research/pull/100/files#diff-be5c63ef3ef39644b98123ee4713f0e5e397efe9d360b210bd2e3e5395f82df9R29 ?
at least the exception is too broad